src/evaluators/BinaryExpression.js

```diff
@@ -63,6 +64,10 @@ export function getPureBinaryOperationResultType(
     let leftPure = purityTest(realm, lval);
     let rightPure = purityTest(realm, rval);
     if (leftPure && rightPure) return typeIfPure;
+    if (lval.getType() === SymbolValue || rval.getType() === SymbolValue) {
```

Comment:
This is probably too aggressive. Not every binary operation on symbols will fail at runtime. The error would be better generated at the actual operation, where there is more context.